ipatests/test_integration/test_ipa_ipa_migration.py

```diff
@@ -872,3 +872,53 @@ def test_ipa_migrate_with_log_file_option(self):
             extra_args=params,
         )
         assert self.replicas[0].transport.file_exists(custom_log_file)
+
+    def test_ipa_migrate_stage_mode_with_cert(self):
+        """
+        This testcase checks ipa-migrate command
+        works without the ValuerError option
+        error when -Z <cert> option is used
+        """
```

Comment:
Just a nit here. So the change evolved because -Z didn't work at all and threw this weird option error. IIRC it's because whatever was stored was invalid or not present. Do we need to test for that case or is returncode 0 sufficient (e.g. the migration was successful)? This reads like a negative test case when it is instead the positive one.

Otherwise the changes LGTM.

Reply:
@rcritten the testcase was for a negative scenario we have invalid cert, i have modified the test description accordingly.

Reply:
But in this case it *is* a valid cert. This is the positive case where migration is successful. It failed previously because the provided cert wasn't being passed into the LDAP connection hence the invalid option. So I think this is more a baseline/ipa-migrate returns no error when running in startTLS mode.
So I don't think you need to check for anything other than returncode == 0. If this were to regress we'd see it right away with this test without doing the message comparision.

Reply:
@rcritten have made the change in testcase to check only for returncode ==0.